1306-x86-sha1.c-add-file.patch

```diff
@@ -0,0 +1,231 @@
+From c90bc4b8f65f70b5ab72dbca5846c8b3dbd94234 Mon Sep 17 00:00:00 2001
+From: Krystian Hebel <krystian.hebel@3mdeb.com>
+Date: Tue, 25 Oct 2022 16:04:17 +0200
+Subject: [PATCH 06/11] x86/sha1.c: add file
+
+File comes from [1] and is licensed under MIT License. Only enough
+changes to make it compile under Xen and to swap endianness of result
+were made to the original file.
```

Comment:
This implementation was chosen for another project that couldn't use GPL, and since it was relatively standalone, we decided to use it here as well. As long as adopting code from Linux won't cause too much work, I think it can be used.

Reply:
Ripping SHA1 and SHA256 out of Linux seems to require untangling some dependencies on macros and moving some code around, so not sure it's worth the trouble.

Reply:
You're right about macros not being a problem, I got hung up on some and didn't check if they are already available in Xen.  I took implementation from Linux after combining parts of 4 files for each hash function.  Commits messages go into more detail.

https://github.com/TrenchBoot/xen/compare/dd475a4a76..aem-phase4-rebase#diff-29310bd3216c70628c5f01dd642e7828d28b9cc148676e56fe5f6b972c261ce7